### PR TITLE
Boost page

### PR DESCRIPTION
**Pull request by YuriyLu**

### CODE REVIEW

File: app/utils/walletManagement.js [hunk 1]
```diff
@@ -242,51 +241,107 @@ export function convertPeerValueToNumberValue(val) {
...
-      user,
-      tokens,
-    },
-    process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
-    true,
-  );
+export function calculateNewBoost(userBoostStat, newStake) {
+  let predictedBoost;
+  let newAverageStake;
+  if (userBoostStat) {
```
Comment (Lampardks) [inline app/utils/walletManagement.js:321]: do a check at the beginning, then you will get rid of one nested if

```
if (!userBoostStat) {
  return [0, null]
}
```

File: app/utils/walletManagement.js [hunk 2]
```diff
@@ -242,51 +241,107 @@ export function convertPeerValueToNumberValue(val) {
...
+    const totalStake = Number(
+      userBoostStat.totalPeriodStake[0].toString() / 10 ** 18,
```
Comment (Lampardks) [inline app/utils/walletManagement.js:323]: you often use this expression, make it a constant 10 ** 18

File: app/utils/walletManagement.js [hunk 3]
```diff
@@ -242,51 +241,107 @@ export function convertPeerValueToNumberValue(val) {
...
+    );
+    const totalUsers = Number(userBoostStat.totalPeriodStake[1].toString());
+    newStake = Number(newStake);
+
+    if (userBoostStat.userStakeNext > 0) {
+      newAverageStake =
+        (totalStake - userBoostStat.userStakeNext + newStake) / totalUsers;
+    } else {
+      newAverageStake = (totalStake + newStake) / (totalUsers + 1);
```
Comment (Lampardks) [inline app/utils/walletManagement.js:332]: number 1 is the magic variable

File: app/utils/walletManagement.js [hunk 4]
```diff
@@ -242,51 +241,107 @@ export function convertPeerValueToNumberValue(val) {
...
+    }
+
+    if (newStake <= newAverageStake) {
+      predictedBoost = (newStake / newAverageStake) * 5 + 1;
```
Comment (Lampardks) [inline app/utils/walletManagement.js:336]: numbers 1 and 5 are the magic variables

File: app/utils/walletManagement.js [hunk 5]
```diff
@@ -242,51 +241,107 @@ export function convertPeerValueToNumberValue(val) {
...
+    } else {
+      predictedBoost = newStake / newAverageStake + 5;
```
Comment (Lampardks) [inline app/utils/walletManagement.js:338]: number 5 is the magic variable

File: app/containers/Boost/Weeks.jsx [hunk 6]
```diff
@@ -32,29 +32,27 @@ const CurrentPendingWeeks = styled.div`
...
         !getWeekStatProcessing && (
           <ul className="mt-3">
             <CurrentPendingWeeks inRow>
               <CurrentWeek
                 locale={locale}
-                {...boostWeeks.currentWeek}
+                period={weekStat[0].period}
+                periodStarted={weekStat[0].periodStarted}
+                periodFinished={weekStat[0].periodFinished}
+                {...userBoostStat}
```
[the remark this answers is not in the document]
Comment (YuriyLu) [reply app/containers/Boost/Weeks.jsx]: Done

File: app/containers/Wallet/View.jsx [hunk 7]
```diff
@@ -21,18 +22,20 @@ const View = ({
...
       account={account}
       balance={balance}
+      availableBalance={availableBalance}
       stakedInCurrentPeriod={stakedInCurrentPeriod}
       stakedInNextPeriod={stakedInNextPeriod}
     />
 
     {userBoostStat &&
-      !userBoostStat.length && <BoostBanner userId={userId} locale={locale} />}
+      !(userBoostStat[0] + userBoostStat[1] > 2) && (
```
[the remark this answers is not in the document]
Comment (YuriyLu) [reply app/containers/Wallet/View.jsx]: Done

File: app/utils/walletManagement.js [hunk 8]
```diff
@@ -242,51 +241,107 @@ export function convertPeerValueToNumberValue(val) {
...
-  );
+export function calculateNewBoost(userBoostStat, newStake) {
+  let predictedBoost;
+  let newAverageStake;
+  if (userBoostStat) {
+    const totalStake = Number(
+      userBoostStat.totalPeriodStake[0].toString() / 10 ** 18,
+    );
+    const totalUsers = Number(userBoostStat.totalPeriodStake[1].toString());
+    newStake = Number(newStake);
```
[the remark this answers is not in the document]
Comment (YuriyLu) [reply app/utils/walletManagement.js]: Done

File: app/utils/walletManagement.js (hunk 1 above)
Comment (YuriyLu) [reply app/utils/walletManagement.js]: Done

File: app/utils/walletManagement.js (hunk 2 above)
Comment (YuriyLu) [reply app/utils/walletManagement.js]: Done

File: app/containers/Wallet/View.jsx [hunk 9]
```diff
@@ -20,33 +20,39 @@ const View = ({
   pickupRewardDispatch,
   pickupRewardProcessing,
   ids,
-}) => (
-  <>
-    <NavHeader userId={userId} />
+}) => {
+  const [currentBoostValue, nextBoostValue] = userBoostStat || [];
+  const BOOSTS_SUM_VALUE_WITHOUT_STAKE = 2;
```
Comment (Lampardks) [inline app/containers/Wallet/View.jsx:23]: take it out of the component

**Pull request merged**